runtime/jcl/common/criusupport.c

```diff
@@ -44,3 +54,94 @@ Java_com_ibm_jvm_CRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *env, jclass unused
 
 	return res;
 }
+
+jint JNICALL
+Java_com_ibm_jvm_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
+							jclass unused,
+							jobject checkPointDir,
+							jboolean keepRunning,
+							jboolean shellJob,
+							jboolean extUnixSupport,
+							jint logLevel,
+							jobject logFile)
+{
+	jint rc = UNSUPPORTED_OPERATION;
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	char directoryBuf[STRING_BUFFER_SIZE] = {0};
+	char *directoryUTF = NULL;
+	char logFileBuf[STRING_BUFFER_SIZE] = {0};
+	char *logFileUTF = NULL;
+
+	if (J9_ARE_ALL_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_ENABLE_CRIU_SUPPORT)) {
+#if defined(LINUX)
+		j9object_t cpDir = NULL;
+		j9object_t log = NULL;
+		IDATA dirFD = 0;
+		PORT_ACCESS_FROM_VMC(currentThread);
+
+		vmFuncs->internalEnterVMFromJNI(currentThread);
+
+		cpDir = J9_JNI_UNWRAP_REFERENCE(checkPointDir);
```

Comment:
@gacholio can you please review this bit.

Reply:
N

Reply:
> the JNI stuff is standard
Well that is a review. :)